### PR TITLE
INT insertion at aggregate switch

### LINKED ISSUE #169: Add INT header creation to aggregate
Need to add the INT Shim logic that currently resides in gateway to the aggregate for the lab trial

### LINKED ISSUE #187: Expand lab_trial data_forward tests
Currently we are only testing that the packets were received by the inet but we also need to inspect the cloned packets (aka. telemetry report too)

### PR DESCRIPTION
**Pull request by ghost**
#### What does this PR do?
Fixes #169 
Fixes #187 
- Adds INT header insertion at the aggregate
- Adds test cases to validate the processing of INT and non-INT packets at the ingress of the aggregate switch.
#### Do you have any concerns with this PR?
No
#### How can the reviewer verify this PR?
Run CI
#### Any background context you want to provide? 
No
#### Questions:
- Have you connected this PR to the issue it resolves? Yes
- Does the documentation need an update? No
- Does this add new dependencies? No
- Have you added unit or functional tests for this PR?

### CODE REVIEW

**Comment by Sanjhana-Jayagopal** [comment]
Run lab trial build

**Review by spisarski — CHANGES_REQUESTED** [review]
While fixing the tofino aggregate tests, please make the two minor formatting changes.

**Comment by Sanjhana-Jayagopal** [comment]
- Added a match key to aggregate's 'add_switch_id_t' to fix issues with Tofino aggregate tests.
- Removed 'device_id' as an action parameter from aggregate's 'data_inspection_t' table.
- Send 'add_switch_id_t' table entry when instantiating a switch.

**Comment by Sanjhana-Jayagopal** [comment]
Run Tofino aggregate build

File: trans_sec/p4runtime_lib/bmv2.py [hunk 1]
```diff
@@ -292,6 +312,20 @@ def add_data_inspection(self, dev_id, dev_mac):
...
             AGG_CTRL_KEY, dev_mac, action_params)
 
+    def add_switch_id(self, dev_id):
+        action_params = {
+            'switch_id': self.sw_info['id']
+        }
+        table_entry = self.p4info_helper.build_table_entry(
+            table_name='{}.add_switch_id_t'.format(self.p4_ingress),
+            match_fields={
+                'hdr.udp.dst_port': 0x022b
```
Comment (spisarski) [inline trans_sec/p4runtime_lib/bmv2.py:322]: Can you use the value consts.UDP_INT_DST_PORT here?

File: playbooks/scenarios/full/packet-flood.yml [hunk 2]
```diff
@@ -37,7 +37,7 @@
     send_src_port: "{{ '%s' % range(2000, 10000) | random(seed=now|int + 2) }}"
     send_packet_count: 150
     min_received_packet_count: 99
-    max_received_packet_count: 150
+    max_received_packet_count: "{{ send_packet_count if send_protocol == 'UDP' else send_packet_count|int*3 }}"
```
Comment (spisarski) [inline playbooks/scenarios/full/packet-flood.yml:40]: Can't do this. You will be masking an issue with dropping packets that currently is not occurring.
Please replace with either 150 or "{{ send_packet_count }}" which I actually like better.

File: playbooks/scenarios/full/packet-flood.yml [hunk 3]
```diff
@@ -66,7 +66,7 @@
     send_src_port: "{{ '%s' % range(2000, 10000) | random(seed=now|int + 4) }}"
     send_packet_count: 150
     min_received_packet_count: 99
-    max_received_packet_count: 150
+    max_received_packet_count: "{{ send_packet_count if send_protocol == 'UDP' else send_packet_count|int*3 }}"
```
Comment (spisarski) [inline playbooks/scenarios/full/packet-flood.yml:69]: Can't do this. You will be masking an issue with dropping packets that currently is not occurring.
Please replace with either 150 or "{{ send_packet_count }}" which I actually like better.

File: playbooks/scenarios/full/packet-flood.yml [hunk 4]
```diff
@@ -95,7 +95,7 @@
     send_src_port: "{{ '%s' % range(2000, 10000) | random(seed=now|int + 6) }}"
     send_packet_count: 150
     min_received_packet_count: 99
-    max_received_packet_count: 150
+    max_received_packet_count: "{{ send_packet_count if send_protocol == 'UDP' else send_packet_count|int*3 }}"
```
Comment (spisarski) [inline playbooks/scenarios/full/packet-flood.yml:98]: Can't do this. You will be masking an issue with dropping packets that currently is not occurring.
Please replace with either 150 or "{{ send_packet_count }}" which I actually like better.

File: playbooks/scenarios/lab_trial/data-inspection.yml [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+          - key: switch_id
+            value: "{{ switch.id }}"
+    data_inspection_table_action_hop2:
+      control_name: TpsCoreIngress
+      table_name: data_inspection_t
+      action:
+        name: "{{ 'data_inspect_packet' }}"
+        match_fields:
+          - key: hdr.udp_int.dst_port
+            value: "{{ '0x022b' }}"
```
Comment (spisarski) [inline playbooks/scenarios/lab_trial/data-inspection.yml:54]: Try and see if the value 555 (without quotes) will work here

**Review by spisarski — CHANGES_REQUESTED** [review]

File: playbooks/scenarios/lab_trial/data_inspection_basic.yml [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+- import_playbook: ../test_cases/send_receive.yml
+  vars:
+    int_hops: "{{ sr1_rec_int_hops | default(0) }}"
+    receiver_log_filename: "di_basic_receiver-{{ send_protocol }}-ip-{{ ip_version }}-ih-{{ int_hops }}-1.out"
+    sender_log_filename: "di_basic_sender-{{ send_protocol }}-ip-{{ ip_version }}-ih-{{ int_hops }}-1.out"
+    send_port: "{{ '%s' % range(2000, 10000) | random(seed=now|int + 9) }}"
+    send_src_port: "{{ '%s' % range(2000, 10000) | random(seed=now|int + 10) }}"
+    send_packet_count: "{{ '%s' % 100 | random(seed=now|int + 11) }}"
+    inspection_data: "{{ sr1_data_inspection_int | default(None) }}"
+    max_received_packet_count: "{{ send_packet_count if send_protocol == 'UDP' else send_packet_count|int*2 }}"
```
Comment (spisarski) [inline playbooks/scenarios/lab_trial/data_inspection_basic.yml:49]: This isn't as bad as what we have in packet-flood.yml, but I believe this is masking an issue with TCP acks.

**Review by spisarski — CHANGES_REQUESTED** [review]
I think we've gotten our TCP ack issue back.

**Review by spisarski — APPROVED** [review]

**Pull request merged**